src/Icon.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Class Icon.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+/**
+ * Icons used to visually represent the state of a validation error.
+ *
+ * @package AmpProject\AmpWP
+ */
+class Icon {
+
+	/**
+	 * Indicates there are validation errors for the AMP page.
+	 */
+	const INVALID = 'amp-invalid';
+
+	/**
+	 * Indicate an AMP version of the page is available.
+	 */
+	const LINK = 'amp-link';
+
+	/**
+	 * Indicates the page is valid AMP.
+	 */
+	const VALID = 'amp-valid';
+
+	/**
+	 * Indicates there are validation errors which have not been explicitly accepted.
+	 */
+	const WARNING = 'amp-warning';
+
+	/**
+	 * Current icon.
+	 *
+	 * @var string
+	 */
+	private static $icon;
+
+	/**
+	 * Invalid icon.
+	 *
+	 * @return Icon
+	 */
+	public static function invalid() {
```

Comment:
The named constructors should not set static state, this can get messy easily. They should just call the regular constructor with the correct argument.

If you want to avoid an open constructor that requires additional validation, you can make the constructor `private`, which will force consumers to pass through one of these named constructors.